### DISCUSSION TITLE
Need help for the correct way to use beforeFormFilled()

### DISCUSSION THREAD

**Discussion by micraux** [filamentphp/filament#13712] [Help]
### Package

Form builder

### Package Version

v3.2.96

### How can we help you?

Hello,

I search in the documentation and the discussions but I think I must be missing something simple here...

I would like to use a `createAction()` inside a RelationManager and pass a value to the form.

So, here is my code:
```php
// OrdersRelationManager.php
public function table(Table $table): Table
{
    return OrderResource::table($table)
        ->headerActions([
            Tables\Actions\createAction::make()
                ->beforeFormFilled(
                    function (array $data): array {
                        $data['action_type'] = 'test';
                        return $data;
                    }
                )
        ]);
}

// OrderResource.php
public static function form(Form $form): Form
{
    return $form
        ->schema([
            Forms\Components\Placeholder::make('action_type'),
            ...
```
I can't figure out how to retrieve the value ("test").

Any help?

**Comment by leandrocfe** [comment]
Placeholder does not have a state. You can use a ViewField instead
https://filamentphp.com/docs/3.x/forms/fields/custom#view-fields

in the view
```html
<div>
    {{ $getState() }}
</div>
```

**Reply by micraux** [reply]
Hello @leandrocfe,

Thank you, but not working... I can see the red background so the view is called, but there is no value in it:
```php
Forms\Components\ViewField::make('action_type')
    ->view('filament.views.simple-viewfield'),

// resources/filament/views/simple-viewfield.blade.php
<div class="bg-red-500 h-8">
    {{ $getState() }}
</div>
```

I choose a Placeholder to see if the value is populated during development, but it will be a Hidden field.

Do I still need to use a ViewField?

BTW, my goal is to pass a parameter to the form and be able to use it in order to perform some custom things.

Do you think it is the right way to do it or is there a better option?

**Reply by leandrocfe** [reply]
Do you mean, relation manager form or resource form? Where is your placeholder?

**Comment by micraux** [comment]
Any idea, someone?

**Comment by micraux** [comment]
Hello @leandrocfe,

My form is in the OrderResource and my OrdersRelationManager calls it.

But I can put it directly in the OrdersRelationManager as I don't want to have the OrderResource appearing in the menu.

Sadly, I just tried to put the form directly in the OrderResourceManager using your ViewField solution, but still not populating the value.

In fact, I want to create orders only from the OrdersRelationManager because I have different types of orders depending of the Client,  hence the `action_type` field.

By the way, I just found (n the documentation about [global research](https://filamentphp.com/docs/3.x/panels/resources/global-search#dispatching-livewire-events-from-global-search-actions) that you can use `->dispatch()` to pass a parameter to a livewire component: could this be a solution for my need?

I am sorry to disturb you and feel terribly stupid, as I am convinced it is a fairy simple thing and I can't find the solution.

**Comment by micraux** [comment]
Hi,

I came to a solution on the Discord channel: thank you Dennis Koch and Pablo Zagni!

I post it here in case someone needs it:

```php
->beforeFormFilled(
    function ($livewire) {
        $livewire->mountedTableActionsData[0]['action_type'] = 'test';
    }
)
->form([
    Forms\Components\TextInput::make('whatever')
        ->default(
            fn (Get $get) => $get('action_type')
        ),
])
```
Big thanks also to @leandrocfe for the help!

**Reply by falconur** [reply]
Hi @micraux.
I have a bit different situation. Can you share something to read about beforeFormFilled if you have found

**Reply by micraux** [reply]
Hi,

I would be happy to help you!

Can you describe your situation (component, relation manager...) and what you are trying to achieve?

**Reply by falconur** [reply]
I need it in RelationManager. I think it does not support this method

**Reply by micraux** [reply]
Well, it can work!
Read again my first message, I was describing my problem:
`I would like to use a createAction() inside a RelationManager and pass a value to the form.`
